core/src/main/java/org/everit/json/schema/NumberSchemaValidatingVisitor.java

```diff
@@ -62,9 +63,9 @@ void visitMaximum(Number maximum) {
         if (maximum == null) {
             return;
         }
-        if (exclusiveMaximum && maximum.doubleValue() <= numberSubject) {
+        if (exclusiveMaximum && compare(maximum, numberSubject) <= 0) {
             owner.failure(subject + " is not less than " + maximum, "exclusiveMaximum");
-        } else if (maximum.doubleValue() < numberSubject) {
+        } else if (compare(maximum, numberSubject) < 0) {
```

Comment:
Can you please extract the `compare(maximum, numberSubject)` value into a local variable?